sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/LiteralSuffixUpperCase.cs

```diff
@@ -48,14 +49,18 @@ protected override void Initialize(SonarAnalysisContext context)
                     var literal = (LiteralExpressionSyntax)c.Node;
                     var text = literal.Token.Text;
 
-                    if (text[text.Length - 1] == 'l' &&
-                        c.SemanticModel.GetTypeInfo(literal).Type.Is(KnownType.System_Int64))
+                    if (text[text.Length - 1] == 'l' && !ShouldIgnore(text))
```

Comment:
don't want to use `text.EndsWith("l")` ?

Reply:
as discussed offline, it's readable as is